Skip redis tests until PHP7.2 stabilizes

markstory opened the pull request:
This will let us get passing builds out of PHP7.2 and hopefully make changes in future beta's more visible.

markstory committed:
Skip redis tests until PHP7.2 stabilizes

This will let us get passing builds out of PHP7.2 and hopefully make
changes in future beta's more visible.

ADmad commented:
Seems the check needs to be tweaked, redis tests were still run on 7.2

antograssiot commented on tests/TestCase/Cache/Engine/RedisEngineTest.php:
```diff
@@ -33,6 +33,7 @@ public function setUp()
     {
         parent::setUp();
         $this->skipIf(!class_exists('Redis'), 'Redis extension is not installed or configured properly.');
+        $this->skipIf(version_compare(PHP_VERSION, '7.2.0beta', '>='), 'Redis is misbehaving in PHP7.2');
```
@markstory the version used is 7.2.0-dev see https://travis-ci.org/cakephp/cakephp/jobs/262358142#L505

markstory replied:
I'll get that updated.

markstory committed:
Check for php7.2dev travis isn't running beta just yet.

ADmad commented:
A new issue has cropped up :( https://travis-ci.org/cakephp/cakephp/jobs/265292163#L658

markstory commented:
And then there is this [lolz build](https://travis-ci.org/cakephp/cakephp/jobs/265292165)

markstory committed:
Revert changes made in 2dbbc08323eac9014c1f9bdc36b2069c0c3081f9

They result in headers_sent() warnings in PHP7.2 as session_ids cannot
be changed there.

Refs #11158

markstory committed:
Is the name $value the problem?

PHP7.2 is doing strange things when variables are named `value`. Perhaps
a different name will fare better.

markstory committed:
Rename more $value variables to see if travis is more happy.

markstory committed:
Rename another $value variable.

Sacrifice to the PHP7.2 gods.

ADmad commented on src/Collection/Iterator/SortIterator.php:
```diff
@@ -65,12 +65,12 @@ public function __construct($items, $callback, $dir = SORT_DESC, $type = SORT_NU
+        foreach ($items as $key => $val) {
+            $val = $callback($val);
+            if ($val instanceof DateTimeInterface && $type === SORT_NUMERIC) {
+                $val = $value->format('U');
```
You missed renaming one instance of $value here.

markstory replied:
Whoops.

markstory commented:
Are folks ok with the variable renames? They seemed to have tricked the tests into passing!
